### PR TITLE
feat(useWebWoker): return worker

### LINKED ISSUE #381: useWebWorker: return also the worker
I use web workers with [Comlink](https://github.com/GoogleChromeLabs/comlink) and cannot use `useWebWorker` function, I need the web worker to proxy it.

### PR DESCRIPTION
**Pull request by wheatjs**
Closes #381

### CODE REVIEW

File: packages/core/useWebWorker/index.ts [hunk 1]
```diff
@@ -55,5 +55,6 @@ export function useWebWorker(
     data,
     post,
     terminate,
+    worker: worker!,
```
Comment (antfu) [inline packages/core/useWebWorker/index.ts:58]: The problem here is that the worker is a variable and might changes during the lifecyle. Better  to make it a shallowRef internally and expose the ref.

Comment (wheatjs) [reply packages/core/useWebWorker/index.ts:58]: Ah yeah I started doing that at first but had some issues with the typing, I'll try to update it to use shallowRef again

**Comment by wheatjs** [comment]
For the types I'm no sure how to keep it consistent with what it was before, so I created a dummy variable. Let me know if you have a better idea on how we should handle that

File: packages/core/useWebWorker/index.ts [hunk 2]
```diff
@@ -22,38 +22,41 @@ export function useWebWorker(
   } = options
 
   const data: Ref<any> = ref(null)
-  let worker: Worker
+  let _worker: Worker
+  const worker = shallowRef<Worker>()
 
-  const post: typeof worker.postMessage = function post(val: any) {
-    if (!worker)
+  const post: typeof _worker.postMessage = function post(val: any) {
```
Comment (antfu) [inline packages/core/useWebWorker/index.ts:28]:
```suggestion
  const post: typeof Worker['postMessage'] = function post(val: any) {
```

Would this work?

Comment (wheatjs) [reply packages/core/useWebWorker/index.ts:28]: Doesn't seem to, gives the following error

```
Property 'postMessage' does not exist on type '{ new (stringUrl: string | URL, options?: WorkerOptions | undefined): Worker; prototype: Worker; }'
```

**Review by antfu — APPROVED** [review]

**Pull request merged**